### PR TITLE
fix(router): fix openapi and add jsonschema validation

### PR DESCRIPTION
**Pull request by OlivierDehaene**
@drbh, can you check the validation part?

### CODE REVIEW

**Review by drbh — APPROVED** [review]
LGTM just tested and worked locally. Thank you!

**Review by drbh — DISMISSED** [review]
LGTM thanks for adding the `draft202012` feat

File: server/text_generation_server/utils/tokens.py [hunk 1]
```diff
@@ -338,10 +337,10 @@ def __call__(
                 _scores = self.repetition_processor(input_ids, _scores)
             if self.frequency_processor is not None:
                 _scores = self.frequency_processor(input_ids, _scores)
-            for warper in self.warpers:
-                _scores = warper(input_ids, _scores)
             if self.grammar_processor is not None:
                 _scores = self.grammar_processor(_scores, self.fsm_grammar_states)
+            for warper in self.warpers:
+                _scores = warper(input_ids, _scores)
```
Comment (OlivierDehaene) [inline server/text_generation_server/utils/tokens.py:343]: @drbh it's very important that logits processor are done BEFORE warpers. Otherwise you can generate nonsense. For example if top_k is set and you warp before the grammar processor, may remove all valid tokens and you cannot generate anything. See https://github.com/huggingface/text-generation-inference/pull/1539#issuecomment-1953392530

Comment (drbh) [reply server/text_generation_server/utils/tokens.py:343]: thank you for pointing out! this is very helpful and makes a lot of sense. Thanks for resolving!
